includes/admin/upgrades/v3/class-data-migrator.php

```diff
@@ -244,13 +244,18 @@ public static function logs( $data = null ) {
 				'date_modified' => $data->post_modified_gmt,
 			);
 
-			$meta_to_remove    = array(
+			$meta_to_remove = array(
 				'_edd_log_file_id',
 				'_edd_log_payment_id',
 				'_edd_log_price_id',
 				'_edd_log_customer_id',
 				'_edd_log_ip',
+				'_edd_log_user_id',
 			);
+			// If the log doesn't have a customer ID, but does have a user ID, keep the user ID as metadata.
+			if ( empty( $log_data['customer_id'] ) && ! empty( $post_meta['_edd_log_user_id'] ) && ! in_array( $post_meta['_edd_log_user_id'], array( 0, -1, '-1' ), true ) ) {
```

Comment:
```suggestion
			if ( empty( $log_data['customer_id'] ) && ! empty( $post_meta['_edd_log_user_id'] ) && ! in_array( $post_meta['_edd_log_user_id'], array( 0, -1 ) ) ) {
```

IMO this is a perfect case for not doing a strict check, because we want to account for both integers and strings.